components/form-toggle/index.js

```diff
@@ -0,0 +1,100 @@
+/**
+ * External dependencies
+ */
+import classNames from 'classnames';
+
+/**
+ * WordPress dependencies
+ */
+import { Component } from 'element';
+
+/**
+ * Internal dependencies
+ */
+import './style.scss';
+
+class FormToggle extends Component {
+	constructor() {
+		super( ...arguments );
+
+		this.onKeyDown = this.onKeyDown.bind( this );
+		this.onClick = this.onClick.bind( this );
+		this.onLabelClick = this.onLabelClick.bind( this );
+	}
+
+	componentWillMount() {
+		this.id = this.constructor.idNum++;
+	}
+
+	onKeyDown( event ) {
+		if ( this.props.disabled ) {
+			return;
+		}
+
+		if ( event.key === 'Enter' || event.key === ' ' ) {
+			event.preventDefault();
+			this.props.onChange();
+		}
+
+		this.props.onKeyDown( event );
+	}
+
+	onClick() {
+		if ( ! this.props.disabled ) {
+			this.props.onChange();
+		}
+	}
+
+	onLabelClick( event ) {
+		if ( this.props.disabled ) {
+			return;
+		}
+
+		const nodeName = event.target.nodeName.toLowerCase();
+		if ( nodeName !== 'a' && nodeName !== 'input' && nodeName !== 'select' ) {
+			event.preventDefault();
+			this.props.onChange();
+		}
+	}
+
+	render() {
+		const id = this.props.id || 'toggle-' + this.id;
+		const wrapperClasses = classNames( 'components-form-toggle__wrapper', {
+			'is-disabled': this.props.disabled,
+		} );
+		const toggleClasses = classNames( 'components-form-toggle', this.props.className );
+
+		// Disable reason: The "checkbox" input is simulated using onClick Events
+		/* eslint-disable jsx-a11y/no-static-element-interactions, jsx-a11y/onclick-has-role, jsx-a11y/click-events-have-key-events */
+		return (
+			<span className={ wrapperClasses }>
+				<input
+					className={ toggleClasses }
+					type="checkbox"
+					checked={ this.props.checked }
+					readOnly={ true }
+					disabled={ this.props.disabled }
+					/>
+				<label className="components-form-toggle__label" htmlFor={ id } >
+					<span className="components-form-toggle__switch"
```

Comment:
>I'm not very familiar with accessibility concerns but I thought that since we're hiding the input, we should provide a "replacement" simulating its behaviour.

In my demo, it's more that the input is visually hidden, but still present on the page so that it can receive and react to focus and keyboard events natively. In the Calypso implementation, this was recreated on a static element with a set of ARIA and role props, `tabIndex` for focus, and keyboard/click handlers. I raise primarily because it seems to me like needless and fragile complexity.